### ISSUE TITLE
[FEATURE REQ] Azure Search FieldBuilder support for complex type

### ISSUE THREAD

**Issue by thdotnet** [Azure/azure-sdk-for-net#7716]
It seems that when trying to use FieldBuilder with a complex type it's not supported. The example in the official docs using the hotel model, translates the addresses to collection of string, but when trying with a complex type, it raises an error:

Cannot create a complex field without sub-fields.

**Comment by brjohnstmsft** [comment]
FieldBuilder does support complex types. What version of Microsoft.Azure.Search are you using?

Also, can you please provide a link to the example you mentioned?

**Comment by brjohnstmsft** [comment]
Also, can you provide a minimal retro of the issue you’re seeing?

**Comment by thdotnet** [comment]
I figure out the problem, but not the rootcase. I've noticed that the version 10.0.0 of Microsoft.Azure.Search.Service is responsible for this part of the code. However, by default I usually only install Microsoft.Azure.Search (which has a dependency for Microsoft.Azure.Search.Service). Without installing manually Microsoft.Azure.Search.Service, it seems to be using an old version. I've noticed this because there's no AnalyzerName.Create anymore.
